File: sn-hd-controler/src/main/webapp/service/knowledgelibService.js
//服务层
app.service('knowledgelibService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../knowledgelib/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../knowledgelib/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../knowledgelib/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../knowledgelib/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../knowledgelib/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../knowledgelib/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../knowledgelib/search.do?page='+page+"&rows="+rows, searchEntity);
	}
	//更新solr数据
    this.updatesolr=function(){
        return $http.post('../knowledgelib/solrupdate.do');
    }
    this.knoSearch=function(searchMap){
        return $http.post('../knowledgelib/knoSearch.do',searchMap);
    }

});

File: sn-search-interface/src/main/java/com/sn/search/KnowledgelibService.java
package com.sn.search;
import com.sn.entity.PageResult;
import com.sn.pojo.Knowledgelib;

import java.util.List;
import java.util.Map;


/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface KnowledgelibService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Knowledgelib> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Knowledgelib knowledgelib);
	
	
	/**
	 * 修改
	 */
	public void update(Knowledgelib knowledgelib);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Knowledgelib findOne(Integer id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Knowledgelib knowledgelib, int pageNum, int pageSize);

	/**
	 * 更新solr数据
	 */
	public void updateSOlr();


	/**
	 * 查询知识库数据
	 * @return
	 */
	List knoSearch(Map searchMap);

	/**
	 * 查询知识库数据总数
	 * @return
	 */
	Long selectCount();
}
